runtime/src/main/java/org/corfudb/runtime/collections/streaming/StreamPollingScheduler.java

```diff
@@ -226,25 +255,54 @@ private void poll(List<StreamingTask> tasks) {
             allQueryResults.putAll(res);
         }
 
-        Preconditions.checkState(tasks.size() == queries.size());
+        // TODO pankti: This validation must be revised
+        //Preconditions.checkState(tasks.size() == queries.size());
 
-        for (int idx = 0; idx < tasks.size(); idx++) {
+        for (int idx = 0; idx < tasks.size(); ) {
             StreamingTask task = tasks.get(idx);
+            StreamAddressSpace sas;
             try {
-                StreamAddressRange taskQuery = queries.get(idx);
-                Preconditions.checkState(task.getStream().getStreamId().equals(taskQuery.getStreamID()));
-                Preconditions.checkState(allQueryResults.containsKey(taskQuery.getStreamID()),
+                if (task instanceof LRStreamingTask) {
+                    sas = getMergedAddressSpace((LRStreamingTask)task, queries, allQueryResults, idx);
+                    idx += ((LRDeltaStream)task.getStream()).getStreamsTracked().size();
```

Comment:
yes that's right.  Good catch.

idx is also used to get the stream query from the flattened list.  Since we now have 2 queries corresponding to a task, the index for the query will have to be incremented differently from the task.  Making the fix.  Will also add a comment explaining this.